src/PyHyperScattering/CMSGIWAXSLoader.py

```diff
@@ -0,0 +1,352 @@
+from PIL import Image
+from PyHyperScattering.FileLoader import FileLoader
+import os, pathlib, datetime, warnings, json
+# import corrections
+import xarray as xr
+import pandas as pd
+#from pyFAI import azimuthalIntegrator
+import numpy as np
+from tqdm.auto import tqdm 
+
+class CMSGIWAXSLoader:
+    """
+    GIXS Data Loader Class | NSLS-II 11-BM (CMS)
+    Used to load single TIFF time-series TIFF GIWAXS images.
+    """
+    def __init__(self, md_naming_scheme=[], root_folder=None):
+        self.md_naming_scheme = md_naming_scheme
+        self.root_folder = root_folder
+        self.sample_dict = None
+        self.selected_series = []
```

Comment:
We should make `delim` an instance variable (see comments below), adding it here.
Also adding a warning if a user doesn't provide a md_naming_scheme.  It's hard to envision this being optional -- might be better to have it just be required -- but I could see wanting to get the xarray in and then sort out coordinates later.
```suggestion
    def __init__(self, md_naming_scheme=[], root_folder=None, delim='_'):
        self.md_naming_scheme = md_naming_scheme
        if len(md_naming_scheme) == 0:
            warnings.warn('Provided an empty md_naming_scheme.  You will not be able to load any metadata in this state.  Set md_naming_scheme to a meaningful value before loading.',stacklevel=2)
        self.root_folder = root_folder
        self.delim = delim
        self.sample_dict = None
        self.selected_series = []
        
```

Reply:
This makes sense. I've been thinking to add an option to load data with just an attribute for the filename, which maybe should be the default if no scheme is provided. Then you'd be able to load data quickly without worrying about naming inconsistencies, but still retrieve the important metadata later on from the saved filename attribute.